integration_test/ops_agent_test.go

```diff
@@ -3069,6 +3085,124 @@ metrics:
 	})
 }
 
+func getRecentServiceOutputForPlatform(platform string) string {
+	if gce.IsWindows(platform) {
+		cmd := strings.Join([]string{
+			"$Past = (Get-Date) - (New-TimeSpan -Minute 1)",
+			"Get-WinEvent -MaxEvents 10 -FilterHashtable @{ Logname='Application'; ProviderName='google-cloud-ops-agent'; StartTime=$Past } | select -ExpandProperty Message",
+		}, ";")
+		return cmd
+	}
+	return "sudo systemctl status google-cloud-ops-agent"
+}
+
+func listenToPortForPlatform(platform string) string {
+	if gce.IsWindows(platform) {
+		cmd := strings.Join([]string{
+			`echo '$Listener = [System.Net.Sockets.TcpListener]20202; $Listener.Start(); Start-Sleep -Seconds 600' > C:\script.ps1`,
+			`Invoke-WmiMethod -Path 'Win32_Process' -Name Create -ArgumentList 'powershell.exe -file C:\script.ps1'`,
+		}, ";")
+
+		return cmd
+	}
+	return "nohup nc -l -p 20202 1>/dev/null 2>/dev/null &"
+}
+
+func TestPortsAndAPIHealthChecks(t *testing.T) {
+	t.Parallel()
+	gce.RunForEachPlatform(t, func(t *testing.T, platform string) {
+		t.Parallel()
+		onlyReadScopes := strings.Join([]string{
+			"https://www.googleapis.com/auth/monitoring.read",
+			"https://www.googleapis.com/auth/logging.read",
+			"https://www.googleapis.com/auth/devstorage.read_write",
+		}, ",")
+		ctx, logger, vm := agents.CommonSetupWithExtraCreateArguments(t, platform, []string{"--scopes", onlyReadScopes})
+
+		if !gce.IsWindows(vm.Platform) {
+			packages := []string{"netcat"}
+			err := agents.InstallPackages(ctx, logger.ToMainLog(), vm, packages)
+			if err != nil {
+				t.Fatalf("failed to install %v with err: %s", packages, err)
+			}
+		}
+
+		if _, err := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", listenToPortForPlatform(vm.Platform)); err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(time.Minute)
```

Comment:
I'm assuming here that it takes a bit for the port to bind? Maybe a brief comment on why we are sleeping for 1 min here, otherwise people will be eager to remove this.

Reply:
I added a comment and changed the sleep to 30 seconds. I added it to pre-emptively avoid flakiness, since all my experiments with ports always had some delays.

Reply:
Addressed in https://github.com/GoogleCloudPlatform/ops-agent/commit/c3a99172b61e046ea0962d76c08554132120f8e3 .